src/CanvasRenderingContext2d.cc

```diff
@@ -2816,16 +2816,23 @@ Context2d::MeasureText(const Napi::CallbackInfo& info) {
 
   metrics = PANGO_LAYOUT_GET_METRICS(layout);
 
+  text_align_t alignment = state->textAlignment;
+
+  // Convert start/end to left/right based on direction
+  if (alignment == TEXT_ALIGNMENT_START) {
+    alignment = (state->direction == "rtl") ? TEXT_ALIGNMENT_RIGHT : TEXT_ALIGNMENT_LEFT;
+  } else if (alignment == TEXT_ALIGNMENT_END) {
+    alignment = (state->direction == "rtl") ? TEXT_ALIGNMENT_LEFT : TEXT_ALIGNMENT_RIGHT;
+  }
```

Comment:
Looks like this would be pretty easy to split out into a function since we do the exact same thing in `setTextPath`. What do you think?

Reply:
Will do, not well versed in c++ but will manage

Reply:
done